intercept_openai_chat_streaming.go

```diff
@@ -161,7 +165,7 @@ func (i *OpenAIStreamingChatInterception) ProcessRequest(w http.ResponseWriter,
 				MsgID:          processor.getMsgID(),
 				Input:          calculateActualInputTokenUsage(lastUsage),
 				Output:         lastUsage.CompletionTokens,
-				Metadata: Metadata{
+				ExtraTokenTypes: map[string]int64{
```

Comment:
BTW is there any generic method to convert ALL metrics from TokenDetails, or do we not want to do this?

Reply:
Not sure I understand the question. Can you rephrase?

Reply:
in pseudo-code:

`json(lastUsage.PromptTokensDetails).unmarshalToMapStringInt64()`

Reply:
We can automatically bring any new metrics an AI provider emits.